src/magnify/registry.py

```diff
@@ -29,11 +32,254 @@ def component_factory(*args, **kwargs):
     return component_decorator
 
 
-def mini_chip(**kwargs):
+def mini_chip(
+    data: ArrayLike | str,
+    times: Sequence[int] | None = None,
+    channels: Sequence[str] | None = None,
+    debug: bool = False,
+    shape: tuple[int, int] = (8, 8),
+    pinlist: str | None = None,
+    blank: str | list[str] | None = None,
+    overlap: int = 102,
+    rotation: int = 0,
+    row_dist: float = 375 / 1.61,
+    col_dist: float = 400 / 1.61,
+    min_button_radius: int = 4,
+    max_button_radius: int = 15,
+    low_edge_quantile: float = 0.1,
+    high_edge_quantile: float = 0.9,
+    num_iter: int = 5000000,
+    min_roundness: float = 0.2,
+    cluster_penalty: float = 50,
+    roi_length: int = 61,
+    progress_bar: bool = False,
+    search_timestep: list[int] | None = None,
+    search_channel: str | list[str] | None = None,
+    squeeze: bool = True,
+    roi_only: bool = False,
+    drop_tiles: bool = True,
+) -> xr.Dataset | list[xr.Dataset]:
+    pipe = mini_chip_pipe(
+        debug=debug,
+        shape=shape,
+        pinlist=pinlist,
+        blank=blank,
+        overlap=overlap,
+        rotation=rotation,
+        row_dist=row_dist,
+        col_dist=col_dist,
+        min_button_radius=min_button_radius,
+        max_button_radius=max_button_radius,
+        low_edge_quantile=low_edge_quantile,
+        high_edge_quantile=high_edge_quantile,
+        num_iter=num_iter,
+        min_roundness=min_roundness,
+        cluster_penalty=cluster_penalty,
+        roi_length=roi_length,
+        progress_bar=progress_bar,
+        search_timestep=search_timestep,
+        search_channel=search_channel,
+        squeeze=squeeze,
+        roi_only=roi_only,
+        drop_tiles=drop_tiles,
+    )
+    return pipe(data=data, times=times, channels=channels)
+
+
+"""
+    Processes image data for a miniaturized chip with button detection, stitching, rotation, and multiple filtering steps.
+
+    Parameters
+    ----------
+    data : 
+        The input image data to be processed. It can be one of the following:
+        - A file path (string) to image data.
+        - An `xarray.DataArray` or `xarray.Dataset` containing image data.
+        - A sequence (list or tuple) of file paths, `xarray.DataArray`, or `xarray.Dataset`.
+        The function can handle multiple formats and will standardize them into an `xarray.Dataset` for processing.
+    times : 
+        A list or sequence of time points to extract from the dataset. If specified, only the data
+        corresponding to these time points will be processed. If `None`, all available time points
+        in the data will be used.
```

Comment:
This isn't true but rather times is a list of names to assign to each timepoint. If you don't specify them and that information isn't available in the metadata of the provided data either then the timepoints will just be initialized to just be 0, 1, 2, ....

It doesn't filter out anything.